spring-kafka/src/test/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainerMockTests.java

```diff
@@ -1116,6 +1118,178 @@ public boolean handleOne(Exception thrownException, ConsumerRecord<?, ?> record,
 		assertThat(recordsDelivered.get(2)).isEqualTo(record1);
 	}
 
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	@Test
+	void pruneRevokedPartitionsFromPendingOutOfOrderCommitsLegacyAssignor() throws InterruptedException {
```

Comment:
Gary, I don't see any `Assignor` settings in the test code.
How does it work, please?

Reply:
It's simulated.

With a legacy assignor, all current assignments are revoked and all (or a subset) are re-assigned. With a coop assignor, only a subset are revoked and (possibly) previously unassigned partitions might be assigned.

See the `case 1:` poll phases in both tests - that is where the simulation is. In the coop case, I test that `consumer.pause()` is called twice (once for the original assignments and once for the newly assigned partition).

Reply:
That's cool. Thank you!